external/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisBackedBlockRDD.scala

```diff
@@ -147,6 +153,17 @@ class KinesisSequenceRangeIterator(
   private var lastSeqNumber: String = null
   private var internalIterator: Iterator[Record] = null
 
+  // variable for kinesis wait time interval between next retry
+  private val kinesisWaitTimeMs = JavaUtils.timeStringAsMs(
+    Try {sparkConf.get("spark.streaming.kinesis.retry.waitTime")}
```

Comment:
It may also be useful to declare these keys as public constants in a sensible location such as the [companion object to ```KinesisInputDStream```](https://github.com/apache/spark/blob/master/external/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisInputDStream.scala#L84), e.g.:

```scala
object KinesisInputDStream {
...
  /**
   * Relevant doc
   */
  val RETRY_WAIT_TIME_KEY = "spark.streaming.kinesis.retry.waitTime"
 
  /**
   * Relevant doc
   */
  val RETRY_MAX_ATTEMPTS_KEY = "spark.streaming.kinesis.retry.maxAttempts"
...
```

This will make things a little less brittle for users who want to dynamically fill in SparkConf values in their apps. You would also be able use these constants in unit tests here.